rwatch/storage/storage_persist.c

```diff
@@ -0,0 +1,243 @@
+/* storage_persist.c
+ * Persist data storage api
+ * libRebbleOS
+ *
+ * Authors: Barry Carter <barry.carter@gmail.com>
+ */
+#include "rebbleos.h"
+#include "rdb.h"
+#include "storage_persist.h"
+#include "appmanager_thread.h"
+
+#define MODULE_NAME "stor"
+#define MODULE_TYPE "RWAT"
+#define LOG_LEVEL RBL_LOG_LEVEL_DEBUG //RBL_LOG_LEVEL_ERROR
+
+#define PERSIST_MAX_STORAGE_BYTES 4096
+
+typedef struct app_persist_key_t {
+    uint32_t appid;
+    uint32_t key;
+} app_persist_key;
+
+bool persist_exists(const uint32_t key)
+{
+    char buffer[1];
+    return persist_read(key, buffer, 1) == 1;
+}
+
+int persist_get_size(const uint32_t key)
+{
+    struct rdb_iter it;
+    rdb_select_result_list head;
+    list_init_head(&head);
+    
+    struct rdb_database *db = rdb_open(RDB_ID_APP_PERSIST);
```

Comment:
I wonder if we need a mechanism to isolate apps from each other (i.e., to prevent an app from filling up the database) -- that is to say, to give each app its own rdb.

Reply:
yes, I agree. The composite key was a quick way to achieve this without epic plumbing

Reply:
(also would be easier at ensuring per-app storage limits)